Feat/fish

bpevs opened the pull request:
# Support fish (#7)

Primary change is addition of fish script for cd:
https://github.com/bpevs/johnny_decimal/pull/8/files#diff-a0c4c51ba70689ec41133321f66f31b836dbecf3176809c32c47ea02d6d1038aR31


Tasks:
  - [x] update install script
  - [x] add fish category cd
  - [x] add fish area cd
  - [x] add fish id cd
  - [x] update uninstall script

Other:
- Pinned STD dependencies to [current](https://deno.land/std@0.152.0) to avoid any unexpected breakage

bpevs committed:
feat: support fish sh

bpevs committed:
fix: supports fish install + cd to id

  - todo: fix area cd
  - todo: fix id cd
  - todo: update uninstall script

brandonpittman reviewed (commented):
@bpevs 

I refactored the Fish script a bit, if you're interested.

```fish
function _jd_navigate
  test -z $argv && cd $JD_HOME && return 1
  cd $argv
end

set _jd_category_regex '^(?<category>[0-9]{2})$' 
set _jd_category_and_id_regex '^(?<category>[0-9]{2})\.(?<id>[0-9]{2,4})$' 

function _jd_find_category
  _jd_navigate (find -E $JD_HOME -regex ".*/$argv .*" -depth 2 -type d) 2>/dev/null;
end

function _jd_find_category_and_id
    set -l found_category (find -E $JD_HOME -regex ".*/$argv[1] .*" -depth 2 -type d) 2>/dev/null;
    _jd_navigate (find -E $found_category -regex ".*/$argv[2] ?.*" -type d) 2>/dev/null;
end

function jd -d "Johnny Decimal CLI"
  set -l SEARCH $argv[1]

  if string match -rq $_jd_category_regex $SEARCH
    # Matches category regex: `dd`. Navigate to category
    _jd_find_category $category
  else if string match -rq $_jd_category_and_id_regex $SEARCH
    # Matches id regex: `dd.dd`. Navigate to id
    _jd_find_category_and_id $category $id
  else if count $argv > /dev/null
    # If there is a non-cd arg, run deno script
    $DENO_DIR/bin/jd $argv
  else
    # No arg for cd. Navigate to root.
    cd $JD_HOME 2>/dev/null;
  end
 end
```

bpevs committed:
feat: simplest fish cd impl

Simplest impl that has all the features
Can focus on perf after fixing uninstall script

bpevs committed:
feat: update uninstall script, bugfix fish

  todo: fix linting, testing

bpevs committed:
fix unit tests

bpevs committed:
fix tests

- adds some ignores to lint rules for now...

bpevs commented:
Cloudflare Pages bot is for docs.  Shouldn't run on this branch. Please Ignore.
